network-api/networkapi/wagtailpages/factory/homepage_features.py

```diff
@@ -21,35 +22,38 @@ class Meta:
     page = SubFactory(WagtailHomepageFactory)
 
 
-class HomepageFeaturedHighlightsFactory(FeaturedFactory):
+class HomepageFeaturedBlogsFactory(FeaturedFactory):
     class Meta:
-        model = HomepageFeaturedHighlights
+        model = HomepageFeaturedBlogs
 
-    highlight = SubFactory(HighlightFactory)
+    blog = SubFactory(BlogPageFactory)
 
 
-class HomepageFeaturedNewsFactory(FeaturedFactory):
+class HomepageFeaturedHighlightsFactory(FeaturedFactory):
     class Meta:
-        model = HomepageFeaturedNews
+        model = HomepageFeaturedHighlights
 
-    news = SubFactory(NewsFactory)
+    highlight = SubFactory(HighlightFactory)
 
 
 def generate(seed):
-    print('Generating Homepage Highlights and News')
+    print('Generating Homepage Blogs and Highlights')
 
     home_page = get_homepage()
 
     reseed(seed)
 
-    featured_highlights = [HighlightFactory.create() for i in range(6)]
-    featured_news = [NewsFactory.create() for i in range(6)]
+    NUM_BLOGS = 4
+    NUM_HIGHLIGHTS = 5
 
-    home_page.featured_highlights = [
-        HomepageFeaturedHighlightsFactory.build(highlight=featured_highlights[i]) for i in range(6)
+    featured_blogs = BlogPage.objects.all()[:NUM_BLOGS]
+    featured_highlights = [HighlightFactory.create() for i in range(NUM_HIGHLIGHTS)]
+
+    home_page.featured_blogs = [
+        HomepageFeaturedBlogsFactory.build(blog=featured_blogs[i]) for i in range(NUM_BLOGS)
     ]
-    home_page.featured_news = [
-        HomepageFeaturedNewsFactory.build(news=featured_news[i]) for i in range(6)
+    home_page.featured_highlights = [
+        HomepageFeaturedHighlightsFactory.build(highlight=featured_highlights[i]) for i in range(NUM_HIGHLIGHTS)
```

Comment:
We're running list comprehension twice here, but we can do the same thing without the `featured_highlights` intermediary:

```
home_page.featured_highlights = [
    HomepageFeaturedHighlightsFactory.build(
        highlight=HighlightFactory.create()
    )
    for i in range(NUM_HIGHLIGHTS)
]
```